app/javascript/app/providers/esp-models-provider/esp-models-provider-reducers.js

```diff
@@ -0,0 +1,17 @@
+export const initialState = {
+  loading: false,
+  loaded: false,
+  data: {},
+  error: false
+};
+
+const setLoading = (loading, state) => ({ ...state, loading });
+const setLoaded = (loaded, state) => ({ ...state, loaded });
+const setError = (state, error) => ({ ...state, error });
```

Comment:
this is something from previous components, but be careful here. The order of state and param need to stay consistent otherwise is gets crazy.